localstack/services/s3/provider_stream.py

```diff
@@ -125,20 +126,17 @@ def put_object(
         key_object.checksum_algorithm = checksum_algorithm
 
         headers = context.request.headers
-        content_sha_256 = context.request.headers.get("x-amz-content-sha256") or ""
-        try:
-            if content_sha_256.startswith("STREAMING-"):
-                # this is a chunked request, we need to properly decode it while setting the key value
-                decoded_content_length = int(headers.get("x-amz-decoded-content-length", 0))
-                key_object.set_value_from_chunked_payload(body, decoded_content_length)
+        if (
+            content_sha_256 := headers.get("x-amz-content-sha256") or ""
+        ) and content_sha_256.startswith("STREAMING-"):
```

Comment:
Same as above, I think the distinction should be done based on the content encoding header if possible.